src/glmtools.jl

```diff
@@ -20,57 +180,157 @@ end
 function mueta{T<:Real}(::CloglogLink, η::T)
     max(eps(T), exp(η) * exp(-exp(η)))
 end
+function inverselink(::CloglogLink, η)
+    expη = exp(η)
+    μ = -expm1(-expη)
+    omμ = exp(-expη)   # the complement, 1 - μ
+    μ, max(realmin(μ), expη * omμ), max(realmin(μ), μ * omμ)
+end
 
 linkfun(::IdentityLink, μ) = μ
 linkinv(::IdentityLink, η) = η
-mueta(::IdentityLink, η) = 1
+mueta(::IdentityLink, η) = one(η)
+inverselink(::IdentityLink, η) = η, one(η), oftype(η, NaN)
 
 linkfun(::InverseLink, μ) = inv(μ)
 linkinv(::InverseLink, η) = inv(η)
 mueta(::InverseLink, η) = -inv(abs2(η))
+function inverselink(::InverseLink, η)
+    μ = inv(η)
+    μ, -abs2(μ), oftype(μ, NaN)
+end
+
+linkfun(::InverseSquareLink, μ) = inv(abs2(μ))
+linkinv(::InverseSquareLink, η) = inv(sqrt(η))
```

Comment:
So the bad news is that my `iscanonical` methods don't work the way that I thought (I know, if I had written unit tests I would have found this out earlier). 
```jl
julia> using DataFrames, GLM

julia> clotting = DataFrame(u = log.([5,10,15,20,30,40,60,80,100]),
                            lot1 = [118,58,42,35,27,25,21,19,18])
9×2 DataFrames.DataFrame
│ Row │ u       │ lot1 │
├─────┼─────────┼──────┤
│ 1   │ 1.60944 │ 118  │
│ 2   │ 2.30259 │ 58   │
│ 3   │ 2.70805 │ 42   │
│ 4   │ 2.99573 │ 35   │
│ 5   │ 3.4012  │ 27   │
│ 6   │ 3.68888 │ 25   │
│ 7   │ 4.09434 │ 21   │
│ 8   │ 4.38203 │ 19   │
│ 9   │ 4.60517 │ 18   │

julia> gm8 = fit(GeneralizedLinearModel, @formula(lot1 ~ 1 + u), clotting, Gamma(), verbose=true)
1: 0.016729715388848446, 0.0021177181489758666
2: 0.016729715178483907, 1.2574304824954926e-8
DataFrames.DataFrameRegressionModel{GLM.GeneralizedLinearModel{GLM.GlmResp{Array{Float64,1},Distributions.Gamma{Float64},GLM.InverseLink},GLM.DensePredChol{Float64,Base.LinAlg.Cholesky{Float64,Array{Float64,2}}}},Array{Float64,2}}

Formula: lot1 ~ 1 + u

Coefficients:
               Estimate   Std.Error  z value Pr(>|z|)
(Intercept)  -0.0165544 0.000927542 -17.8476   <1e-70
u             0.0153431 0.000414958  36.9751   <1e-99


julia> gm8.model.rr
GLM.GlmResp{Array{Float64,1},Distributions.Gamma{Float64},GLM.InverseLink}([118.0, 58.0, 42.0, 35.0, 27.0, 25.0, 21.0, 19.0, 18.0], Distributions.Gamma{Float64}(α=1.0, θ=1.0), [0.00160669, 0.00746689, 0.00240188, 0.000843898, 0.00147963, 1.23783e-6, 0.000823452, 0.00141054, 0.000695503], [0.00813941, 0.0187744, 0.0249955, 0.0294095, 0.0356306, 0.0400445, 0.0462656, 0.0506796, 0.0541033], [122.859, 53.2639, 40.0071, 34.0026, 28.0658, 24.9722, 21.6143, 19.7318, 18.4832], Float64[], [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0], [15094.3, 2837.04, 1600.57, 1156.18, 787.688, 623.611, 467.179, 389.345, 341.628], [0.000321911, -0.00166938, -0.0012451, -0.000862636, 0.00135305, -4.45692e-5, 0.00131496, 0.00187963, 0.00141432])

julia> GLM.iscanonical(gm8.model.rr)
false
```

The good news is that the apparently redundant calculations in the working weights are not being short circuited and the desired answer was obtained.

I still need to figure out why the methods for `iscanonical` weren't providing the answers I thought they would and how to properly evaluate the working weights.